src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -616,8 +711,18 @@ public CloudApplication getApplication(UUID applicationGuid) {
 
     @Override
     public InstancesInfo getApplicationInstances(String applicationName) {
-        CloudApplication application = getApplication(applicationName);
-        return getApplicationInstances(application);
+        UUID applicationGuid = getRequiredApplicationGuid(applicationName);
+        ApplicationState applicationState = getApplicationResponseByName(applicationName).getState();
+        return applicationState == ApplicationState.STARTED ? findApplicationInstances(applicationGuid) : null;
+    }
+
+    private GetApplicationResponse getApplicationResponseByName(String applicationName) {
+        ApplicationResource applicationResource = getApplicationResourceByName(applicationName).block();
```

Comment:
Doesn't `getApplicationResourceByName` call `applicationsV3().get()`?
And even if not, doesn't the resource(s) it returns contain the same information?

Reply:
Fixed